src/Core/Managed/Shared/Extensibility/Implementation/Metrics/MetricTerms.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Metrics
+{
+    using System;
+
+    internal static class MetricTerms
+    {
+        private const string MetricPropertiesNamePrefix = "Microsoft.ApplicationInsights.Metrics";
+
+        public static class Aggregation
+        {
+            public static class Interval
+            {
+                public static class Moniker
+                {
+                    public const string Key = MetricPropertiesNamePrefix + ".Aggregation.IntervalMs";
+                }
+            }
+        }
+
+        public static class Extraction
+        {
+            public static class ProcessedByExtractors
+            {
+                public static class Moniker
+                {
+                    public const string Key = MetricPropertiesNamePrefix + ".Extraction.ProcessedByExtractors";
+                    public const string ExtractorInfoTemplate = "(Name:{0}, Ver:{1})";      // $"(Name:{ExtractorName}, Ver:{ExtractorVersion})"
+                }
+            }
+        }
+
+        public static class Autocollection
+        {
+            public static class Moniker
+            {
+                public const string Key = MetricPropertiesNamePrefix + ".MetricIsAutocollected";
```

Comment:
It seems to me that given all the other goo AI counts, this is a drop in the ocean. Name-qualifying auto-added properties is a good way to avoid collisions with user properties. In future versions we will need to add new metric documents to support things like percentiles etc. Than we can move this into the internal context..